### ISSUE TITLE
Single channel support for optical flow

### ISSUE THREAD

**Issue by kmtm** [pytorch/vision#5960]
### 🚀 The feature

Would love to be able to run black and white images through the new RAFT optical flow model! (Same goes for ResNet)

### Motivation, pitch

I have a B+W dataset. Duplicating channels feels wasteful.

**Comment by NicolasHug** [comment]
Hi @kmtm 

For now our models only support 3 channels as input. Supporting both 3 and 1 channels would mean that we need to pass that as input to the model's constructor, which is something we'd like to avoid. We would also need to provide different sets of pre-trained weights.
I'm afraid your best course of action here is to duplicate the grey channel, as you already suggested.

**Comment by pmeier** [comment]
> Duplicating channels feels wasteful.

It is, but not by much. Only the very first layer needs to do three times the work, which, due to parallelism, does not translate to 3x the runtime.

> I'm afraid your best course of action here is to duplicate the grey channel, as you already suggested.

That is a very naive solution and likely will give you sub-par results. Although duplicating grayscale for three channels looks "normal" to the human eye, this is a completely different representation for models that were trained on RBG images. You will be better off to replace the layer with a proper grayscale one.

**Comment by NicolasHug** [comment]
I don't know for other datasets but IIRC Kitti contains mostly grayscale images which we converted to RGB during training, so I'm not so sure about your last point.

**Comment by pmeier** [comment]
Yeah, my comment is maybe not as clear as I thought it was. My point is: if the model was trained on RGB images, you will get sub-par results if you have grayscale images and "convert" them to RGB during inference. If you train on this pseudo-RGB representation, using the same during inference is fine.

If we already do that, my point above can be ignored.
